src/js/utils/MetronomeUtil.js

```diff
@@ -1,70 +1,79 @@
 /**
- * Parse an item into a Job or JobTree - This method will create sub trees if
- * needed to insert the item at the correct location
- * (based on id/path matching).
- * @param {{
- *          id:string,
- *          items:array<({id:string, items:array}|*)>,
- * }} parent tree to add item to
- * @param {{id:string}} item job to add to parent
- * @param {[type]} jobsAlreadyAdded hash of id and jobs of jobs that
- * have already been added to the parent
+ * Get the parent id (e.g. foo) form the iten id (e.g. foo.bar) by matching
+ * everything but the first bit including the preceding dot "." (e.g. ".bar")
+ *
+ * @param {string} itemId
+ * @return {string} parent id
  */
-function addJob(parent, item, jobsAlreadyAdded) {
-  const { id } = parent;
+function getParentId(itemId) {
+  const [parentId = ""] = itemId.match(/.*?(?=\.?[^.]+\.?$)/);
```

Comment:
I can't make sense of this line, sorry 😩 
Suggestion: please give that regex a name like `const SECOND_PART_BLA = /.*?(?=\.?[^.]+\.?$)/`